### PR TITLE
React to refactoring of IR phases in Razor

### PR DESCRIPTION
**Pull request by ajaybhargavb**
React to https://github.com/aspnet/Razor/pull/996

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor.Host/InjectDirective.cs [hunk 1]
```diff
@@ -20,13 +20,13 @@ public static IRazorEngineBuilder Register(IRazorEngineBuilder builder)
             return builder;
         }
 
-        internal class Pass : IRazorIRPass
+        internal class Pass : IRazorDirectiveClassifierPass
         {
             public RazorEngine Engine { get; set; }
 
-            public int Order => RazorIRPass.DirectiveClassifierOrder;
+            public int Order => 50;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/InjectDirective.cs]: Why isn't it 0?

**HINT**

I'm going to ask this question every time I see a non-default value. Do yourself a favor and always make a comment in the code before sending a PR.

File: src/Microsoft.AspNetCore.Mvc.Razor.Host/ModelDirective.cs [hunk 2]
```diff
@@ -56,14 +56,14 @@ private static string GetModelType(DocumentIRNode document, Visitor visitor)
...
         }
 
-        internal class Pass : IRazorIRPass
+        internal class Pass : IRazorDirectiveClassifierPass
         {
             public RazorEngine Engine { get; set; }
 
             // Runs after the @inherits directive
-            public int Order => RazorIRPass.DirectiveClassifierOrder + 5;
+            public int Order => 75;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/ModelDirective.cs]: YAY

File: src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperPass.cs [hunk 3]
```diff
@@ -11,21 +11,21 @@
 
 namespace Microsoft.AspNetCore.Mvc.Razor.Host
 {
-    public class ViewComponentTagHelperPass : IRazorIRPass
+    public class ViewComponentTagHelperPass : IRazorIROptimizationPass
     {
         public RazorEngine Engine { get; set; }
 
-        public int Order => RazorIRPass.LoweringOrder;
+        public int Order => RazorIRPass.DefaultFeatureOrder;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperPass.cs]: This should be 0

File: test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs [hunk 4]
```diff
@@ -220,7 +220,7 @@ private DocumentIRNode CreateIRDocument(RazorEngine engine, RazorCodeDocument co
                 var phase = engine.Phases[i];
                 phase.Execute(codeDocument);
 
-                if (phase is IRazorIRPhase)
+                if (phase is IRazorDirectiveClassifierPhase)
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/InjectDirectiveTest.cs]: <3

File: test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/MvcViewDocumentClassifierPassTest.cs [hunk 5]
```diff
@@ -207,7 +207,7 @@ private static RazorCodeDocument CreateDocument(string content, string filePath
 
         private static DocumentIRNode CreateIRDocument(RazorEngine engine, RazorCodeDocument codeDocument)
         {
-            var phases = engine.Phases.TakeWhile(p => !(p is IRazorIRPhase));
+            var phases = engine.Phases.TakeWhile(p => !(p is IRazorDocumentClassifierPhase));
```
Comment (rynowak) [inline test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/MvcViewDocumentClassifierPassTest.cs]: Why not just make this look like the others

**Comment by rynowak** [comment]
Do a control +  F for `"DefaultDirectiveIRPass"`

I had to add some hacks that look at this pass based on it's name. You should be able to remove those hacks now.

File: src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperPass.cs (hunk 3 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperPass.cs]: Q: Why not have everything based around 0 and have the `DefaultFeatureOrder` be 0?

Comment (ajaybhargavb) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperPass.cs]: I think having `DefaultFeatureOrder` be a higher value than 0 is useful especially in cases like `DefaultDocumentClassifierPass`. Mvc and RazorPages uses their own classifiers and they shouldn't have to use negative numbers to override the Order.

Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/ViewComponentTagHelperPass.cs]: The name is bad. We should probably just delete this. We shouldn't be referencing it anywhere in MVC.

**Comment by ajaybhargavb** [comment]
🆙 📅

File: src/Microsoft.AspNetCore.Mvc.Razor.Host/InjectDirective.cs [hunk 6]
```diff
@@ -62,8 +58,6 @@ public DocumentIRNode Execute(RazorCodeDocument codeDocument, DocumentIRNode irD
 
                     visitor.Class.Children.Add(member);
                 }
-
-                return irDocument;
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor.Host/InjectDirective.cs]: Should we also decouple this code from `@model`?

**Pull request merged**